### PR TITLE
Added ScaleNorm as the first MAT layer

### PR DESCRIPTION
**Pull request by atreyamaj**
# Pull Request Template

## Description
This is the first layer for MAT: ScaleNorm
## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by atreyamaj** [comment]
Screenshot from 2021-07-15 15-07-24

File: deepchem/models/torch_models/layers.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+
+class LayerNorm(nn.Module):
+  """Apply Layer Normalization to input.
+
+    The layer takes input and applies layer normalization to it.
+
+    References
+    ----------
+    .. [1] Lukasz Maziarka et al. "Molecule Attention Transformer" Graph Representation Learning workshop and Machine Learning and the Physical Sciences workshop at NeurIPS 2019. 2020. https://arxiv.org/abs/2002.08264
```
Comment (vsomnath) [inline deepchem/models/torch_models/layers.py:15]: Are you sure LayerNorm was discovered in this paper? Its a more general thing

Comment (atreyamaj) [reply deepchem/models/torch_models/layers.py:16]: Thank you, I checked the equation again and other than slight differences it's the same. I have changed it to ScaleNorm now.

File: deepchem/models/torch_models/layers.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  import torch
+  import torch.nn as nn
+except:
+  raise ImportError('These classes require Torch to be installed.')
+
+
+class ScaleNorm(nn.Module):
+  """Apply Scale Normalization to input.
+
+    All G values are initialized to sqrt(d).
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:12]: Can you add more docs here explaining the class and giving a usage example? In general you want a solid paragraph along with a good usage example

File: deepchem/models/torch_models/layers.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    References
+    ----------
+    .. [1] Lukasz Maziarka et al. "Molecule Attention Transformer" Graph Representation Learning workshop and Machine Learning and the Physical Sciences workshop at NeurIPS 2019. 2020. https://arxiv.org/abs/2002.08264
+    """
+
+  def __init__(self, scale, eps=1e-5):
+    """Initialize a ScaleNorm layer.
+
+        Parameters
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:22]: Formatting here is off. Check that you have the right version of `yapf` (0.22). You might need to manually fix

**Review by rbharath — COMMENTED** [review]
Couple of minor comments and should be good to merge!

File: deepchem/models/torch_models/layers.py (hunk 2 above)
Comment (atreyamaj) [reply deepchem/models/torch_models/layers.py]: I have changed the layer explanation and added a usage example like I did for the MATFeaturizer. Is the updated explanation enough or should I try to add a bit more?

File: deepchem/models/torch_models/layers.py (hunk 3 above)
Comment (atreyamaj) [reply deepchem/models/torch_models/layers.py]: I am using `yapf 0.22`. I've fixed the docstrings manually.

**Comment by atreyamaj** [comment]
I've made some doc changes and rebased the branch.

cc: @rbharath @vsomnath

File: deepchem/models/torch_models/layers.py (hunk 2 above)
Comment (vsomnath) [reply deepchem/models/torch_models/layers.py]: You can add an additional line mentioning why / where `ScaleNorm` would be useful?

Comment (atreyamaj) [reply deepchem/models/torch_models/layers.py]: I've added a bit for that now.

**Comment by rbharath** [comment]
Failures don't look related so going to go ahead and merge in

**Pull request merged**